python/bindings/exec.cpp

```diff
@@ -0,0 +1,84 @@
+#include <pybind11/functional.h>
+#include <pybind11/numpy.h>
+#include <pybind11/pybind11.h>
+#include <pybind11/stl.h>
+
+#include "reffine/utils/utils.h"
+
+using namespace std;
+using namespace reffine;
+
+namespace py = pybind11;
+
+string to_string(shared_ptr<Func> fn) { return IRPrinter::Build(fn); }
+
+template <typename Output, typename... Inputs>
+Output execute_query(void* query, Output output, Inputs... inputs)
```

Comment:
read the next comment first before this one.

Reply:
once you have made the assumption that all functions are going to be with only void* args, you don't need a templated function to execute the query. you just need a varrgs function.